builtin/mainmenu/content/contentdb_api.lua

```diff
@@ -0,0 +1,58 @@
+--Minetest
+--Copyright (C) 2023 rubenwardy
+--
+--This program is free software; you can redistribute it and/or modify
+--it under the terms of the GNU Lesser General Public License as published by
+--the Free Software Foundation; either version 2.1 of the License, or
+--(at your option) any later version.
+--
+--This program is distributed in the hope that it will be useful,
+--but WITHOUT ANY WARRANTY; without even the implied warranty of
+--MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+--GNU Lesser General Public License for more details.
+--
+--You should have received a copy of the GNU Lesser General Public License along
+--with this program; if not, write to the Free Software Foundation, Inc.,
+--51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+if not core.get_http_api then
+	return
+end
+
+contentdb_api = {}
+
+
+-- encodes for use as URL parameter or path component
+local function urlencode(str)
+	return str:gsub("[^%a%d()._~-]", function(char)
+		return ("%%%02X"):format(char:byte())
+	end)
+end
+assert(urlencode("sample text?") == "sample%20text%3F")
+
+
+local function fetch_latest_releases(param)
+	local version = core.get_version()
+	local base_url = core.settings:get("contentdb_url")
+	local url = base_url ..
+			"/api/updates/?type=mod&type=game&type=txp&protocol_version=" ..
+			core.get_max_supp_proto() .. "&engine_version=" .. param.urlencode(version.string)
```

Comment:
consider replacing this with a helper that works as follows sometime:
```lua
local url = build_cdb_url("/api/updates", {
    type = {"mod", "game", "txp"}
    protocol_version = core.get_max_supp_proto(),
    engine_version = version.string
})
```

Reply:
Yeah, as a more general function would be cool to have `url_set_query`